core/lib/libtomcrypt/ed25519.c

```diff
@@ -135,9 +139,13 @@ TEE_Result crypto_acipher_ed25519ctx_sign(struct ed25519_keypair *key,
 
 	memzero_explicit(&private_key, sizeof(private_key));
 
-	if (err != CRYPT_OK)
+	if (err != CRYPT_OK && err != CRYPT_BUFFER_OVERFLOW)
 		return TEE_ERROR_BAD_PARAMETERS;
+
 	*sig_len = siglen;
+	if (err == CRYPT_BUFFER_OVERFLOW)
```

Comment:
I could modify the check to:
`if (sig_len == 0 && err == CRYPT_BUFFER_OVERFLOW)`
and only then update the *sig_len value and return `TEE_ERROR_SHORT_BUFFER` but if there is better place to handle it I can update the changes.

The issue I had that I was not able to find a way how to distinguish this case in `crypto_acipher_ed25519_sign()`

Reply:
I think your latter proposal will better apply to OP-TEE and prevent local hacks in libtomcrypt implementation. Preferably with `UL(64)` or straight `64` instead of `64uL`.
I would apply the same change also in `crypto_acipher_ed25519_sign()`, for consistency.